pkg/config/app/kuma-dp/config.go

```diff
@@ -116,8 +131,40 @@ type Dataplane struct {
 	DrainTime config_types.Duration `json:"drainTime,omitempty" envconfig:"kuma_dataplane_drain_time"`
 }
 
+func (d *Dataplane) PostProcess() error {
+	if err := validateMeshOrName(".Name", d.Name); err != nil {
+		podName, ok := os.LookupEnv("POD_NAME")
+		if !ok {
+			return nil
+		}
+
+		podNamespace, ok := os.LookupEnv("POD_NAMESPACE")
+		if !ok {
+			return nil
+		}
+
+		d.Name = fmt.Sprintf("%s.%s", podName, podNamespace)
```

Comment:
Should you redo validateMeshOrName here?

Reply:
addressed